web/app.js

```diff
@@ -646,30 +575,151 @@ const PDFViewerApplication = {
-      );
+  async run(config) {
+    this.preferences = new Preferences();
+    await this.initialize(config);
+
+    const { appConfig, eventBus } = this;
+    let file;
+    if (typeof PDFJSDev === "undefined" || PDFJSDev.test("GENERIC")) {
+      const queryString = document.location.search.substring(1);
+      const params = parseQueryString(queryString);
+      // file param is preferred as it's the one used by upstream, but view is
+      // what CommunityQ is currently coded for.
+      if (params.get("file")) {
+        file = params.get("file") ?? AppOptions.get("defaultUrl");
+      } else if (params.get("view")) {
+        file = params.get("view") ?? AppOptions.get("defaultUrl");
+      }
+      validateFileURL(file);
+    } else if (PDFJSDev.test("MOZCENTRAL")) {
+      file = window.location.href;
+    } else if (PDFJSDev.test("CHROME")) {
+      file = AppOptions.get("defaultUrl");
     }
+    window.currentFile = file;
+
+    // iOS 16.3 workaround
+    if (unsupportedIosPresent()) {
+      document.getElementById("mainBody").classList.add("hiddenView");
+      return;
+    }
+
+    if (typeof PDFJSDev === "undefined" || PDFJSDev.test("GENERIC")) {
+      const fileInput = (this._openFileInput = document.createElement("input"));
+      fileInput.id = "fileInput";
+      fileInput.hidden = true;
+      fileInput.type = "file";
+      fileInput.value = null;
+      document.body.append(fileInput);
+
+      fileInput.addEventListener("change", function (evt) {
+        const { files } = evt.target;
+        if (!files || files.length === 0) {
+          return;
+        }
+        eventBus.dispatch("fileinputchange", {
+          source: this,
+          fileInput: evt.target,
+        });
+      });
+
+      // Enable dragging-and-dropping a new PDF file onto the viewerContainer.
+      appConfig.mainContainer.addEventListener("dragover", function (evt) {
+        evt.preventDefault();
+
+        evt.dataTransfer.dropEffect =
+          evt.dataTransfer.effectAllowed === "copy" ? "copy" : "move";
+      });
+      appConfig.mainContainer.addEventListener("drop", function (evt) {
+        evt.preventDefault();
+
+        const { files } = evt.dataTransfer;
+        if (!files || files.length === 0) {
+          return;
+        }
+        eventBus.dispatch("fileinputchange", {
+          source: this,
+          fileInput: evt.dataTransfer,
+        });
+      });
+    }
+
+    if (!AppOptions.get("supportsDocumentFonts")) {
+      AppOptions.set("disableFontFace", true);
+      this.l10n.get("pdfjs-web-fonts-disabled").then(msg => {
+        console.warn(msg);
+      });
+    }
+
+    if (!this.supportsPrinting) {
+      appConfig.toolbar?.print?.classList.add("hidden");
+      appConfig.secondaryToolbar?.printButton.classList.add("hidden");
+    }
+
+    // Commented by Tim
+    // if (!this.supportsFullscreen) {
+    //   appConfig.secondaryToolbar?.presentationModeButton.classList.add(
+    //     "hidden"
+    //   );
+    // }
```

Comment:
Re-commented this, not sure if we still want it commented. Could probably just change the code in general to do what we want?